src/components/ContactsAdd.js

```diff
@@ -11,29 +22,122 @@ function ContactsAdd(props) {
   //TODO: Implement controlled form
   //send POST to json server on form submit
 
-  return (
-    <form className="form-stack contact-form">
-      <h2>Create Contact</h2>
+  const [newContactForm, setNewContactForm] = useState(initialFormState)
+  const navigate = useNavigate()
+
+
+
+  const handleChange = (event) => {
+    const name = event.target.name;
+    const value = event.target.value;
+
+    
+
+    const newFormState = {...newContactForm }
+    if (name === "firstName") {
+      newFormState.firstName = value
+    }
+
+    if (name === "lastName") {
+      newFormState.lastName = value
+    }
+
+    if (name === "street") {
+      newFormState.street = value
+    }
 
-      <label htmlFor="firstName">First Name</label>
-      <input id="firstName" name="firstName" type="text" required />
+    if (name === "city") {
+      newFormState.city = value
+    }
 
-      <label htmlFor="lastName">Last Name:</label>
-      <input id="lastName" name="lastName" type="text" required/>
+    if (name === "email") {
+      newFormState.email = value
+    }
 
-      <label htmlFor="street">Street:</label>
-      <input id="street" name="street" type="text" required/>
+    if (name === "linkedIn") {
+      newFormState.linkedIn = value
+    }
 
-      <label htmlFor="city">City:</label>
-      <input id="city" name="city" type="text" required/>
+    if (name === "twitter") {
+      newFormState.twitter = value
+    }
+    setNewContactForm(newFormState)
```

Comment:
Exactly how I did it, until I remembered Carlo's quick example he mentioned -> because our initialFormState AND the name variables in the form are identical, we can just do this: newFormState[name] = value
Which will do lines 37-63 all in a single statement, which I thought was kinda neat!